oauth2/handler.go

```diff
@@ -156,9 +156,23 @@ func (h *handler) serveAuthorizationRequest(
 	// redirect_uri
 	if redir, ok := query["redirect_uri"]; ok {
 		for _, r := range client.Redirects {
-			if redir[0] == r {
-				authzState.RedirectURI = r
-				break
+			lastChar := r[len(r)-1:]
+			if lastChar == "*" {
+				length := len(r) - 1
+				if len(redir[0]) >= length {
```

Comment:
Redir[0] is een niet erg omschrijvende naam. 
Beter iets in de geest van:
```
var redirectRequestedURL = redir[0]
var allowedRedirectPattern = r
```

Reply:
I agree and will change the original authors mistake ;)